File: src/main/java/net/zymen/retryablerequests/RequestToChecksumService.java
package net.zymen.retryablerequests;

import org.springframework.stereotype.Service;

import javax.servlet.http.HttpServletRequest;

@Service
class RequestToChecksumService {

    String requestToChecksum(final HttpServletRequest request) {

        return request.getRequestURI();
    }
}

File: retryable-requests-core/src/main/kotlin/net/zymen/retryablerequests/RequestToChecksumService.kt
package net.zymen.retryablerequests;

import net.zymen.retryablerequests.annotations.RetryableRequest
import net.zymen.retryablerequests.retryableby.RetryableBy
import org.springframework.stereotype.Service

@Service
class RequestToChecksumService {

    fun requestToChecksum(request: MyRequestWrapper, annotation: RetryableRequest): String = when (annotation.retryableBy) {
        RetryableBy.QUERY_PARAMS -> request.requestURI + request.queryString
        RetryableBy.BODY -> request.requestURI + request.body
    }
}
